src/app/auth/services/auth.service.ts

```diff
@@ -1,31 +1,59 @@
 import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { User, UserRegistration } from '@shared/models/user.model';
+import { BehaviorSubject, map, Observable } from 'rxjs';
+import { SessionStorageService } from '@app/auth/services/session-storage.service';
+import { Router } from '@angular/router';
+
+interface LoginResponse {
+  result: string;
+}
 
 @Injectable({
   providedIn: 'root',
 })
 export class AuthService {
-  login(user: any) {
-    // replace 'any' with the required interface
+  private API_URL = 'http://localhost:4000/';
+  private isAuthorized$$ = new BehaviorSubject(false);
+  public isAuthorized$ = this.isAuthorized$$.asObservable();
+
+  constructor(
+    private sessionService: SessionStorageService,
+    private http: HttpClient,
+    private router: Router
+  ) {
+    const token = this.sessionService.getToken();
+
+    this.isAuthorized$$.next(!!token);
+  }
+
+  login(user: User): Observable<void> {
     // Add your code here
+    return this.http.post<LoginResponse>(`${this.API_URL}login`, user).pipe(
+      map(response => {
+        if (response.result) {
+          this.sessionService.setToken(response.result);
+        }
+      })
+    );
   }
 
   logout() {
-    // Add your code here
+    this.sessionService.deleteToken();
+    this.isAuthorized$$.next(false);
+    this.router.navigate(['/login']);
   }
 
-  register(user: any) {
-    // replace 'any' with the required interface
-    // Add your code here
+  register(user: UserRegistration) {
+    return this.http.post(`${this.API_URL}register`, user);
   }
 
-  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-  // @ts-ignore
   get isAuthorised() {
-    // Add your code here. Get isAuthorized$$ value
+    return this.isAuthorized$$.getValue();
```

Comment:
I think you need to return the public member of this class here.

Reply:
well, if i try to return the public member then ts will complain about the return type:  
Type Observable<boolean> is not assignable to type boolean

i think that's because the get keyword, so how should i return the public member if the public member is an observable?